examples/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect.ino

```diff
@@ -0,0 +1,591 @@
+  {
+    logString = "Failed to get MQTT username. Return code: ";
+    LogError(logString + result);
+    exit(result);
+  }
+
+  generateMQTTPassword(); // SAS Token
+
+  mqttClient.setId(mqttClientId);
+  mqttClient.setUsernamePassword(mqttUsername, mqttPassword);
+  mqttClient.onMessage(onMessageReceived); // Set callback for C2D messages
+
+  logString = "MQTT Client ID: ";
+  LogInfo(logString + mqttClientId);
+  logString = "MQTT Username: ";
+  LogInfo(logString + mqttUsername);
+  logString = "MQTT Password (SAS Token): ";
+  LogInfo(logString + "***");
+
+  LogInfo("MQTT client initialized.");
+}
+
+/*
+ * connectMQTTClientToAzureIoTHub:
+ * The SSL library sets a callback to validate the server certificate.
+ * The MQTT client connects to the provided hostname. The port is pre-set.
+ * The MQTT client subscribes to the Cloud to Device (C2D) topic to receive messages.
+ */
+void connectMQTTClientToAzureIoTHub() 
+{
+  LogInfo("Connecting to Azure IoT Hub.");
+
+  // Set a callback to get the current time used to validate the server certificate.
+  ArduinoBearSSL.onGetTime(getTime);
+
+  while (!mqttClient.connect(IOT_CONFIG_IOTHUB_FQDN, AZ_IOT_DEFAULT_MQTT_CONNECT_PORT)) 
+  {
+    int code = mqttClient.connectError();
+    logString = "Cannot connect to Azure IoT Hub. Reason: ";
+    LogError(logString + mqttErrorCodeName(code) + ", Code: " + code);
+    delay(5000);
+  }
+
+  LogInfo("Connected to your Azure IoT Hub!");
+
+  mqttClient.subscribe(AZ_IOT_HUB_CLIENT_C2D_SUBSCRIBE_TOPIC);
+
+  logString = "Subscribed to MQTT topic: ";
+  LogInfo(logString + AZ_IOT_HUB_CLIENT_C2D_SUBSCRIBE_TOPIC);
+}
+
+/*------------------------------------------------*/
+/*    Telemetry and message-callback functions    */
+/*------------------------------------------------*/
+
+/*
+ * onMessageReceived:
+ * The function called when device receives a message on the subscribed C2D topic.
+ * Callback function signature is defined by the ArduinoMQTTClient library.
+ * Message received is printed to the terminal.
+ */
+void onMessageReceived(int messageSize) 
+{
+  logString = "Message received: Topic: ";
+  LogInfo(logString + mqttClient.messageTopic() + ", Length: " + messageSize);
+  LogInfo("Message: ");
+
+  while (mqttClient.available()) 
+  {
+    Serial.print((char)mqttClient.read());
+  }
+  Serial.println();
+}
+
+/*
+ * sendTelemetry:
+ * The Azure IoT SDK for C client creates the MQTT topic to publish a telemetry message.
+ * The MQTT client creates and sends the telemetry mesage on the topic.
+ */
+static void sendTelemetry() 
+{
+  digitalWrite(LED_PIN, HIGH);
+  LogInfo("Arduino Nano RP2040 Connect sending telemetry . . . ");
+
+  int result = az_iot_hub_client_telemetry_get_publish_topic(
+      &azIoTHubClient, NULL, telemetryTopic, sizeof(telemetryTopic), NULL);
+  if (az_result_failed(result)) 
+  {
+    logString = "Failed to get telemetry publish topic. Return code: ";
+    LogError(logString + result);
+    exit(result);

```

Comment:
Same comment for exit. The correct implementation should be getting into an infinite loop, which is then debuggable in platforms that so support it.